src/core/utils/channel_manager.cpp

```diff
@@ -89,7 +109,41 @@ void ChannelManager::RequestChannelChange(uint8_t aChannel)
 exit:
     return;
 }
+#endif
+
+#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE
+void ChannelManager::ChangeCslChannel(uint8_t aChannel)
+{
+    if (!(!Get<Mle::Mle>().IsRxOnWhenIdle() && Get<Mac::Mac>().IsCslEnabled()))
+    {
+        // cannot select or use other channel
+        ExitNow();
+    }
+
+    if (aChannel == Get<Mac::Mac>().GetCslChannel())
+    {
+        LogInfo("Already operating on the requested channel %d", aChannel);
+        ExitNow();
+    }
+
+    VerifyOrExit(Radio::IsCslChannelValid(aChannel));
+
+    if (aChannel == Get<Mac::Mac>().GetPanChannel())
+    {
+        LogInfo("Requested channel %d is same as PanChannel, CslChannel removed.", aChannel);
+        aChannel = 0;
```

Comment:
Should we in this case still use the `aChannel` directly?

Setting it to zero means, we dont have explicitly selected CSL channel and device can use the PAN channel for it , but in this case, we may want it to be `aChannel` even if the PAN channel later get changed?